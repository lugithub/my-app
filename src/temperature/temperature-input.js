import React from 'react';

const scaleNames = {
  c: 'Celsius',
  f: 'Fahrenheit'
};

export class TemperatureInput extends React.Component {
  render() {    
    const scale = this.props.scale;
    const labelText = 'ggg';
    return (
      <fieldset>
        <legend>Enter temperature in {scaleNames[scale]}:</legend>
        <input 
          aria-label={labelText}
          value={this.props.temperature}
          onChange={this.props.handleChange} />
      </fieldset>
    );
  }
}
