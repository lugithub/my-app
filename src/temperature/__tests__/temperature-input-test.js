import React from 'react';
import renderer from 'react-test-renderer';

import { TemperatureInput } from '../temperature-input';

it('renders correctly', () => {
    const tree = renderer
      .create(<TemperatureInput temperature={1010}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});