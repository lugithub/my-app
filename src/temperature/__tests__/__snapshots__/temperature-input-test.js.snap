// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<fieldset>
  <legend>
    Enter temperature in 
    :
  </legend>
  <input
    aria-label="ggg"
    value={1010}
  />
</fieldset>
`;
